### ISSUE TITLE
Remove Stetic Gtk#2 GUI designer 

### ISSUE THREAD

**Issue by lytico** [dotdevelop/dotdevelop#48]
see: https://github.com/mono/monodevelop/pull/5486

follow his advice

**Comment by hwthomas** [comment]
I agree.  Sometime, a Gtk#3 designer might be good, but not just now.  The paragraph about moving towards porting MonoDevelop to Gtk#3 is interesting (and relevant)

**Comment by Adahel** [comment]
> see: [mono#5486](https://github.com/mono/monodevelop/pull/5486)
> 
> follow his advice

I partially disagree. Libglade's successor is Glade3 and there is a transition version called Glade 3.8 which depends on GTK+ 2.24 and supports libglade. As of Glade 3.10, legacy support for libglade has been removed and depends on GTK+ 3.0 and above. As DotDevelop is switching to GTK#3 and not GTK4, I am in favor of a slow and smooth transition from Stetic GUI Designer using Glade 3.8 for GTK#2 projects, when SGD is fully compatible with Glade 3.8 it is easy to remove legacy support and support Glade 3.10 and higher. I believe that a slow and smooth transition is a great source of knowledge and is better than writing from scratch something that at first can be hack and unstable. There are people who still want to program SDG and GTK#2 and I think it is unfair to remove the SDG now, although obsolete, fix bugs and receive new features while preparing for GTK#3, I believe it is the best way. I recommend studying SDG and Glade 3.8 to assess the viability of my comment and make the best decision. My comment shows a lot of work and there probably won't be many who agree, unfortunately. If Glade3 is not used, I agree with the total removal of the SDG. If Glade3 is used, get ready for a lot of debugging! :).

Recommended reading:
[The Glade DL](https://blogs.gnome.org/tvb/2011/01/15/the-glade-dl/)

**Comment by Gankov** [comment]
We has one large progject and few small with Static Disign and gtk2. I want have way to smooth transition from Stetic. But i think it is not posible. We have not resurces to rewrite all application at once. Now we rewrite to MVVM, for more fast swithing on new UI. it's a pity that Stetic will be removed, this mean that we cannot support legacy code and projects with dotDevelop.

**Comment by lytico** [comment]
you need stetic or need only glade support?

see this discussion:

https://github.com/GtkSharp/GtkSharp/issues/169
https://github.com/GtkSharp/GtkSharp/issues/169#issuecomment-644769443

its planned to integrate 

https://github.com/GtkSharp/GtkSharp/tree/develop/Source/Addins/MonoDevelop.GtkSharp.Addin

**Comment by Gankov** [comment]
> you need stetic or need only glade support?
I meant Stetic, for work with legagy gtk2 projects. But we can use last Monodevelop for this.
Now i play with GtkSharp, but still not use in production code.

> its planned to integrate
i think it good.

**Comment by Adahel** [comment]
> it's a pity that Stetic will be removed, this mean that we cannot support legacy code and projects with dotDevelop.

Nothing has been decided yet, I think. We are commenting on an issue to find the best way. I agree with you: a smooth transition for me is the best way. With patience and dedication, the Stetic GUI Designer Glade3 is certainly possible. Stetic GUI Designer as an open alternative to WinForms Designer and GTKBuilder from Glade3 as an open alternative to WPF. This is what I want as a distant future for DotDevelop.

**Comment by knocte** [comment]
I agree with all the arguments that advocate for removal, however, I think you can get all the effects that a removal would give you, without impacting any current users: just by disabling the addin by default (current users that still like/use it, can enable it on their end).

**Comment by Adahel** [comment]
I agree with the @knocte, as long as a project that requires the Designer to open contains a message box or something that suggests activating the addin. If the DotDevelop team is going to abandon the Stetic GUI Designer, at least add later a Designer based on the Glade Interface Designer with a Stetic to Glade converter and support for GTK#2 and GTK#3. Why GTK#2? Support for Glade 3.8 and a smooth transition to GTK#3. Why incorporate the GID? Visual Studio does not have a separate Designer and a separate Designer impairs productivity and makes programming difficult for beginners. Scripts can help with productivity, but nothing replaces the ease of a full IDE. 

**My comments are only suggestions, I am not the owner of the truth!**

**Comment by knocte** [comment]
On top of this, Jo's PR (https://github.com/mono/monodevelop/pull/5486) was really done having VS4Mac in mind, not MonoDevelop or any Linux IDE experience whatsoever.

**Comment by hwthomas** [comment]
As I understand it, however, the problem is in building stetic without error, rather than just disabling the addin.  The build needs glade-sharp.dll, which is not easily available now (correct me if I'm wrong), otherwise building stetic throws up about 45 errors.

**Comment by Adahel** [comment]
> The build needs glade-sharp.dll, which is not easily available now (correct me if I'm wrong)...

I don't know if the history of glade-sharp will be useful, but it's on this link: [gtk-sharp/glade](https://github.com/mono/gtk-sharp/commits/master/glade).

**Comment by lytico** [comment]
the main problem is that we have 
no gtk2#-library with `targetframework`  `.netstandard` or any other  that can be consumed by a (dotnetcore)/net5/net5ff - application.
And the plan is to migrate to a net5ff-application in a long term.

But, i guess, we'll have a release based on 8.6 -  `targetframework`  `net471` or `net48`. 

Proposal: As long as gtk2# can be consumed, the addin stays.

**Comment by knocte** [comment]
> Proposal: As long as gtk2# can be consumed, the addin stays.

And even when it stops being able to be consumed, I think it's pointless to remove the source code; you can just disable it in the build system (if we don't delete it, there are more chances that someone can look at it to try to port it to Gtk#4 or whatever).

**Comment by hwthomas** [comment]
If Stetic is going to be left in the build for now, where can ``glade-sharp.dll`` be sourced from? I've had a copy on my production PC for quite a few years (in ``/usr/lib/mono/gac``), but I've no idea where it came from (possibly an earlier install of monodevelop, or it may have originally been packaged with ``gtk-sharp-2.0`` at some time - I really don't know). However, I'm trying to keep the 'Setting up a Build Environment' Wiki pages updated, and ``glade-sharp`` isn't available in any of the main distros.  Any help with where to find this would be appreciated.

**Comment by lytico** [comment]
@hwthomas see: https://github.com/GtkSharp/GtkSharp/issues/169#issuecomment-753965973

**Comment by knocte** [comment]
Or much simpler, in ubuntu install `libglade2.0-cil-dev` package.

**Comment by hwthomas** [comment]
@lytico; @knocte;  Many thanks for that.  I've obviously been looking in the wrong places!
